### ISSUE TITLE
16 Bit Deflate tiff Image Corrupt

### ISSUE THREAD

**Issue by faheemziker** [python-pillow/Pillow#3832]
### What did you do?

i just opened 16bit deflate tiff files with Pillow Image.open function
and when i saved the file it just save the corrupted image

```python
im=Image.open("deflate_tif_1.tif")
im.save("bad_image_1.jpg")
```

### What did you expect to happen?

it should save the correct/same image

### What actually happened?

it is saving corrupted image

### What are your OS, Python and Pillow versions?

* OS: ubuntu

* Python: 2.7
* Pillow: 6.0.0

Sample Images

Deflate tiff (input file)
https://www.dropbox.com/s/hlm50c6z99hh9d2/deflate_tif_1.tif?dl=0

Bad Image produces
https://www.dropbox.com/s/csewdhjz2dhitek/bad_image_1.jpg?dl=0

```python

im=Image.open("deflate_tif_1.tif")
im.save("bad_image_1.jpg")

```

**Comment by faheemziker** [comment]
@radarhere 
can you help me..?

**Comment by kkopachev** [comment]
It's probably same as https://github.com/python-pillow/Pillow/issues/3770

**Comment by faheemziker** [comment]
@kkopachev 
I have the latest master with support of old-style JPEG compression
but still having bad image

**Comment by kkopachev** [comment]
@faheemziker you could use Pillow 5.4.1 which does not have this problem. 

This issue is a duplicate of #3770. 

I opened PR to fix it: #3837
